src/app/routes/storage.ts

```diff
@@ -285,6 +289,268 @@ export class StorageController {
     });
   }
 
+  public async moveFile(req: Request, res: Response): Promise<void> {
+    const { behalfUser: user } = req as SharedRequest;
+    const { fileId, destination } = req.body;
+
+    if (Validator.isInvalidString(fileId)) {
+      throw createHttpError(400, 'File ID is not valid');
+    }
+
+    if (Validator.isInvalidPositiveNumber(destination)) {
+      throw createHttpError(400, 'Destination folder ID is not valid');
+    }
+
+    const clientId = String(req.headers['internxt-client-id']);
+
+    return this.services.Files.MoveFile(user, fileId, destination)
+      .then(async (result: { result: FileAttributes }) => {
+        res.status(200).json(result);
+        const workspaceMembers = await this.services.User.findWorkspaceMembers(user.bridgeUser);
+        workspaceMembers.forEach(
+          ({ email }: { email: string }) => void this.services.Notifications.fileUpdated({
+            file: result.result,
+            email: email,
+            clientId: clientId
+          }),
+        );
+      })
+      .catch((err: Error) => {
+        this.logger.error(err);
+        res.status(500).json({
+          error: err.message
+        });
+      });
+  }
+
+  public async updateFile(req: Request, res: Response): Promise<void> {
+    const { behalfUser: user } = req as SharedRequest;
+    const fileId = req.params.fileid;
+    const { metadata, bucketId, relativePath } = req.body;
+    const mnemonic = req.headers['internxt-mnemonic'];
+    const clientId = String(req.headers['internxt-client-id']);
+
+    if (Validator.isInvalidString(fileId)) {
+      throw createHttpError(400, 'File ID is not valid');
+    }
+
+    if (Validator.isInvalidString(bucketId)) {
+      throw createHttpError(400, 'Bucket ID is not valid');
+    }
+
+    if (Validator.isInvalidString(relativePath)) {
+      throw createHttpError(400, 'Relative path is not valid');
+    }
+
+    if (Validator.isInvalidString(mnemonic)) {
+      throw createHttpError(400, 'Mnemonic is not valid');
+    }
+
+    return this.services.Files.UpdateMetadata(user, fileId, metadata, mnemonic, bucketId, relativePath)
+      .then(async (result: FileAttributes) => {
+        res.status(200).json(result);
+        const workspaceMembers = await this.services.User.findWorkspaceMembers(user.bridgeUser);
+        workspaceMembers.forEach(
+          ({ email }: { email: string }) => void this.services.Notifications.fileUpdated({
+            file: result,
+            email,
+            clientId
+          }),
+        );
+      })
+      .catch((err: Error) => {
+        this.logger.error(`Error updating metadata from file ${fileId} : ${err}`);
+        res.status(500).json(err.message);
+      });
+  }
+
+  public async deleteFileBridge(req: Request, res: Response): Promise<void> {
+    if (req.params.bucketid === 'null') { // Weird checking...
```

Comment:
For God's sake

Reply:
idk.. i wanted to modify the logic inside the handler as little as possible, so I just left it was

Reply:
should i change it? So far it works..

Reply:
I left this comment just as a thought of this legacy code that we would change once Desktop old users upgrade from 1.5. The goal - which has been accomplished very good - was to left anything as it is but in TypeScript, do not worry